bundles/config/org.eclipse.smarthome.config.core/src/main/java/org/eclipse/smarthome/config/core/status/ConfigStatusInfo.java

```diff
@@ -0,0 +1,170 @@
+        return filter(parameterNamesCollection, new Predicate<ConfigStatusMessage>() {
+            @Override
+            public boolean apply(ConfigStatusMessage configStatusMessage) {
+                return parameterNamesCollection.contains(configStatusMessage.parameterName);
+            }
+        });
+    }
+
+    /**
+     * Adds the given {@link ConfigStatusMessage}.
+     *
+     * @param configStatusMessage the configuration status message to be added
+     *
+     * @throws IllegalArgumentException if given configuration status message is null
+     */
+    public void add(ConfigStatusMessage configStatusMessage) {
+        if (configStatusMessage == null) {
+            throw new IllegalArgumentException("Config status message must not be null");
+        }
+        configStatusMessages.add(configStatusMessage);
+    }
+
+    /**
+     * Adds the given given {@link ConfigStatusMessage}s.
+     *
+     * @param configStatusMessages the configuration status messages to be added
+     *
+     * @throws IllegalArgumentException if given collection is null
+     */
+    public void add(Collection<ConfigStatusMessage> configStatusMessages) {
+        if (configStatusMessages == null) {
+            throw new IllegalArgumentException("Config status messages must not be null");
+        }
+        for (ConfigStatusMessage configStatusMessage : configStatusMessages) {
+            add(configStatusMessage);
+        }
+    }
+
+    private Collection<ConfigStatusMessage> filter(Collection<?> filter, Predicate<ConfigStatusMessage> predicate) {
+        if (filter.isEmpty()) {
+            return getConfigStatusMessages();
+        }
+        return Collections.unmodifiableCollection(Collections2.filter(getConfigStatusMessages(), predicate));
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see java.lang.Object#hashCode()
+     */
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((configStatusMessages == null) ? 0 : configStatusMessages.hashCode());
+        return result;
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see java.lang.Object#equals(java.lang.Object)
+     */
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj)
+            return true;
+        if (obj == null)
+            return false;
+        if (getClass() != obj.getClass())
+            return false;
+        ConfigStatusInfo other = (ConfigStatusInfo) obj;
+        if (configStatusMessages == null) {
+            if (other.configStatusMessages != null)
+                return false;
+        } else if (!configStatusMessages.equals(other.configStatusMessages))
+            return false;
+        return true;
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see java.lang.Object#toString()
+     */
+    @Override
+    public String toString() {
+        return "ConfigStatusInfo [configStatusMessages=" + configStatusMessages + "]";
```

Comment:
ok, I didn't actively test it - so if you say it spits out all the messages from the array then all is fine.